src/services/__tests__/anchor-service.test.ts

```diff
@@ -281,6 +281,42 @@ describe('ETH service',  () => {
     expect(requests.length).toEqual(0)
   });
 
+  test('Unlimited anchor requests', async () => {
```

Comment:
This test passed with no code changes, so this doesn't actually trigger the bug.  I really have no idea why this worked in tests but failed when run for real.  I don't even understand why the code changes I did here are necessary or fixed anything, but I can definitely see the behavior change in manual testing...